bf2fc6cc711aee1a0c2a/kas-fleet-manager
# change api base path

## Description

Also changes error prefix to KAFKAS_MGMT. Closes https://issues.redhat.com/browse/MGDSTRM-2374 and https://issues.redhat.com/browse/MGDSTRM-3380. 

This is done in two commits so that the second commit https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/commit/0d1be12b884ccda9766fe396efa03652834cbd73 can be reverted after 4 weeks once the support for `/api/managed-services-api` is removed. This will be done in https://issues.redhat.com/browse/MGDSTRM-3465

Please also review: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/19437

## TODO
1. ~~Add new path to ocm gateway by updating AppInterface~~ https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/19437
2. ~~Create a JIRA to remove support for `/api/managed-services-api/`  path by reverting the second commit.~~ https://issues.redhat.com/browse/MGDSTRM-3465 
3. Send an email about the change and ask the client side to change to new path: Include the https://issues.redhat.com/browse/MGDSTRM-3466 JIRA  it in the email.


## Verification Steps


1. Make sure that integration tests pass for the new base path
2. Checkout the changes locally and run `make run` to launch the API. 
3. Run the steps in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/README.md#creating-a-kafka-cluster to make sure that the old base path continues to work as expected pass
4. Run the steps in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/5574ab29338c7b5e1e0ae91b4671f6c7d2c84b33/README.md#creating-a-kafka-cluster to make sure that the new code works as expected
5. Verify that errors are working are backward compatible too: 
 - Run `curl --silent -XGET -H "Authorization: Bearer $(ocm token)" http://localhost:8000/api/kafkas_mgmt/v1/errors | jq` and observe the error prefix for the new api path
 - Run `curl --silent -XGET -H "Authorization: Bearer $(ocm token)" http://localhost:8000/api/managed-services-api/v1/errors | jq` and observe the output for the old api path.
 - Run `curl -v -XPOST -H "Authorization: Bearer $(ocm token)" http://localhost:8000/api/managed-services-api/v1/kafkas?async=true -d '{ "region": "us-east-1", "cloud_provider": "aws",  "name": "serviceapi", "multi_az":false} | jq
` and observe the generate error for the old api base path
 - Run `curl -v -XPOST -H "Authorization: Bearer $(ocm token)" http://localhost:8000/api/kafkas_mgmt/v1/kafkas?async=true -d '{ "region": "us-east-1", "cloud_provider": "aws",  "name": "serviceapi", "multi_az":false} | jq
` and observe the generate error for the new api base path
6. Observe the openapi endpoints
- curl --silent -XGET -H "Authorization: Bearer $(ocm token)" http://localhost:8000/api/managed-services-api/v1/openapi | jq
- curl --silent -XGET -H "Authorization: Bearer $(ocm token)" http://localhost:8000/api/kafkas_mgmt/v1/openapi | jq

To make sure that the two paths have appear appropriate for each Open API spec



## Type of change


## Checklist (Definition of Done)

- [x] JIRA has created for changes required on the client side: https://issues.redhat.com/browse/MGDSTRM-3466
